lib/x11test.pm

```diff
@@ -608,8 +608,7 @@ sub firefox_check_default {
     my $stilltime = get_var('TIMEOUT_SCALE') ? 100 : 10;
     wait_still_screen $stilltime, 100;
     # Set firefox as default browser if asked
-    assert_screen [qw(firefox_default_browser firefox-url-loaded)], 300;
-    if (match_has_tag('firefox_default_browser')) {
+    if (check_screen('firefox_default_browser')) {
```

Comment:
this change suppose to be covered by http://10.67.17.178/tests/547/modules/gnome_default_applications/steps/1/src   module but actually haven't reach this code due to soft failure , right ?

Reply:
on run which you mention from osd all seems to be fine . First it match default browser dialog and clicking `use firefox as default` and later check that url loaded with `firefox_check_popups` method and tag `firefox-url-loaded`. Can you show me run from o3 which you meant ?
My understanding that `firefox-url-loaded` should match **only** when no popups is blocking browser and it is "ready to use". In such definition the flow that skiping needles with tag `firefox_default_browser` in case tag `firefox-url-loaded` is matched makes perfect sense because the fact that needle `firefox-url-loaded` is matched is proving that there no popup window with question about default browser.
But take my opinion with grain of salt I am not big expert in this flavor of tests so I might missing something obvious ...

Reply:
Thank you for replying!

Yes the osd case seems to be fine but that is the specified run from osd I would like to emphasize. 

So from what I'm seeing from `lib/x11test.pm`, `firefox-url-loaded` can be matched **no matter there are popups or not**. For example, [this local run](http://10.67.17.178/tests/544#step/gnome_classic_switch/18). 

So regarding the intention of `firefox-url-loaded`, in this case it is used to assert that firefox successfully starts ---- in the case where there is no pop, asserting this tag assures that firefox does start.

In the specified osd run, we can find that `firefox-url-loaded` is checked but not matched. If you could check the needle pictures, the tag is supposed to match in this case -- it did not match because needles pictures are outdated in their fonts ---- this is the reason the osd case went fine. Has this tag matched, `firefox_default_browser` will be skipped thus leave us the pop up un-clicked. 

So it's just how I understand this script and it seems to [be written long ago with even more tag matchings](https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3243/files#diff-cddbbf2a65a923eafb87c32290618fcaR470). 
@asmorodskyi  What do think?

Reply:
make sense , let's see how it will work ...